homeassistant/components/sensor/forecast.py

```diff
@@ -124,80 +116,66 @@ def state(self):
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
-        return self._unit_of_measurement
+        self.forecast_data.update_unit_of_measurement(self.type)
```

Comment:
best to make [this](https://github.com/home-assistant/home-assistant/pull/2217/files/19ed6369c6fde6e395029e81c99c67daaaae9e60#diff-f39240498111d4ab02be098713ba9d96R237) **return** the unit of measurement and save it in the sensor's `__init__`